pkg/backup/rbac.go

```diff
@@ -0,0 +1,154 @@
+/*
+Copyright 2018 the Heptio Ark contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package backup
+
+import (
+	arkdiscovery "github.com/heptio/ark/pkg/discovery"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	rbac "k8s.io/api/rbac/v1"
+	rbacbeta "k8s.io/api/rbac/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	rbacclient "k8s.io/client-go/kubernetes/typed/rbac/v1"
+	rbacclientbeta "k8s.io/client-go/kubernetes/typed/rbac/v1beta1"
+)
+
+type CRBLister interface {
+	// List returns a slice of ClusterRoleBindings which can represent either v1 or v1beta1 ClusterRoleBindings.
+	List() ([]ClusterRoleBinding, error)
+}
+
+type V1CRBLister struct {
+	client rbacclient.ClusterRoleBindingInterface
+}
+
+func (v1 V1CRBLister) List() ([]ClusterRoleBinding, error) {
+	var crbs []ClusterRoleBinding
+
+	crbList, err := v1.client.List(metav1.ListOptions{})
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+	for _, crb := range crbList.Items {
+		crbs = append(crbs, v1crb{crb: crb})
+	}
+
+	return crbs, nil
+}
+
+type V1beta1CRBLister struct {
+	client rbacclientbeta.ClusterRoleBindingInterface
+}
+
+func (v1beta1 V1beta1CRBLister) List() ([]ClusterRoleBinding, error) {
+	var crbs []ClusterRoleBinding
+
+	crbList, err := v1beta1.client.List(metav1.ListOptions{})
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+	for _, crb := range crbList.Items {
+		crbs = append(crbs, v1beta1crb{crb: crb})
+	}
+
+	return crbs, nil
+}
+
+// NewCRBLister wraps the v1beta1 and v1 RBAC client's List with a common interface.
+func NewCRBLister(log logrus.FieldLogger, clientset kubernetes.Interface) (CRBLister, error) {
+	var lister CRBLister
+	discoveryHelper, err := arkdiscovery.NewHelper(clientset.Discovery(), log)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+	var supportedAPI metav1.GroupVersionForDiscovery
+	apigroups := discoveryHelper.APIGroups()
+	for _, ag := range apigroups {
+		if ag.Name == "rbac.authorization.k8s.io" {
+			supportedAPI = ag.PreferredVersion
+			break
+		}
+	}
+
+	switch supportedAPI.Version {
+	case "v1":
```

Comment:
could use `rbac.SchemeGroupVersion.Version` rather than "v1" here, and `rbacbeta.`... for the v1beta1  case